### ISSUE TITLE
[REVIEW]: Kactus2: A graphical EDA tool built on the IP-XACT standard

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#151]
**Submitting author:** @TermosPullo (<a href="http://orcid.org/0000-0003-2254-3332">Janne Virtanen</a>)
**Repository:** <a href="https://github.com/kactus2/kactus2dev" target ="_blank">https://github.com/kactus2/kactus2dev</a>
**Version:** 3.4.0
**Editor:** @labarba
**Reviewer:** @jflorence
**Archive:** <a href="http://dx.doi.org/10.5281/zenodo.570521" target="_blank">10.5281/zenodo.570521</a>

## Status

[![status](http://joss.theoj.org/papers/73e33d6850d24f0d6aad0d5f38937f83/status.svg)](http://joss.theoj.org/papers/73e33d6850d24f0d6aad0d5f38937f83)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/73e33d6850d24f0d6aad0d5f38937f83"><img src="http://joss.theoj.org/papers/73e33d6850d24f0d6aad0d5f38937f83/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/73e33d6850d24f0d6aad0d5f38937f83/status.svg)](http://joss.theoj.org/papers/73e33d6850d24f0d6aad0d5f38937f83)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/kactus2/kactus2dev" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

### Conflict of interest

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (such as being a major contributor to the software).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/kactus2/kactus2dev">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (3.2.173)?
- [x] **Authorship:** Has the submitting author (TermosPullo) made major contributions to the software?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** Have any performance claims of the software been confirmed?

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g. API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g. papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS. @jflorence it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As as reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all JOSS reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by labarba** [comment]
The pre-review issue mentions that we (the JOSS editors) are unclear whether this submission satisfies the submission guidelines. 

In particular we require that (quoting from http://joss.theoj.org/about#author_guidelines):

> - Your software should have a research application
>   -Should be a significant contribution to the available open source software that either enables some new research challenges to be addressed or makes addressing research challenges significantly better (e.g., faster, easier, simpler)

Here is the author's reply to that question:
https://github.com/openjournals/joss-reviews/issues/117#issuecomment-260583610

@jflorence : Please help us with a comment providing your assessment of this question here.

**Comment by TermosPullo** [comment]
Hello

It appears that the management decided to close our original repository at SourceForge after we migrated to GitHub: https://github.com/kactus2/kactus2dev. 

On the other hand, it appears that you are still not quite sure if the submission is eligible for the journal. Can we help you with this somehow?

**Comment by jflorence** [comment]
I am still reviewing the software.
Sorry for the delay.
Thank you for letting us know about the change of repository.

**Comment by labarba** [comment]
@TermosPullo : I was going to tell @jflorence that he can open issues on the code repo to report any problems with installation, etc., when I had a quick look at the repo and see that it has issues disabled. Is there a particular reason to disallow issue reporting?

**Comment by TermosPullo** [comment]
Our project management, including issue tracker is concentrated at our redmine service: https://kactus2.cs.tut.fi/

Please notice that it requires registeration and logging in to file issues.

**Comment by jflorence** [comment]
@TermosPullo I tried to create an account there a while ago, but I don't think I was successful.

If I try to create it again, it says that the email address is already taken, but if I try to recover my password, I don't receive any email...
My address is jacques.florence@gatech.edu

**Comment by TermosPullo** [comment]
@jflorence Your account is activated. If you cannot log in, please contact the project management with email: kactus2@cs.tut.fi

**Comment by labarba** [comment]
@jflorence Hi there—would you give us an update here of what your assessment is so far? I see that you've been ticking off items on the review checklist, yet some remain. Can you summarize the status of this review? Thanks!

**Comment by jflorence** [comment]
@labarba @TermosPullo Hello,
Here is my assessment so far:
I would grade the software as MINOR REVISIONS needed.
Mostly, those revisions are in the documentation.
The detail of those can be found as issues 336, 337, 338, 339, 340 on the SW's issues tracker https://kactus2.cs.tut.fi/issues/


Overall, the software meets the criteria of the JOSS.

To address the specific concern you had about whether the SW fits the scope ("Your software should have a research application -Should be a significant contribution to the available open source software that either enables some new research challenges to be addressed or makes addressing research challenges significantly better"):

Kactus2 is a design environment for electronic IP cores, system-on-chips and boards, as well as associated software.
Kactus2 indeed has a research as well as industrial application, as it allows, for instance, better communication of an electronic system design. It can not only accelerate the design itself, but also improve transfer of knowledge between research groups.

Now, about the specific points (reviewer questions at the top):

Conflict of interest:
+I confirm that there are no conflicts of interest for me to review this work.

General Checks:
+The software has a repository.
+The license is GNU GPL2. It is also available as dual-licensing for commercial use.
-The version I reviewed was 3.3.59 and not 3.2.173. Is this a problem?
+Authorship: As per the git log, TermosPullo (JanneVirtanen) has made significant contributions to the software.

Functionality
+Installation: Installation went smoothly
+Functionality: Functionality is as described
+Performance: No performance claims made

Documentation:
-A statement of need: The authors clearly state what problems the software is designed to solve and who the target audience is. However, this is done on the website funbase.cs.tut.fi and not in the software itself.
+Installation instruction: The dependencies are clearly stated
-Example usage: Examples cannot be seen straightforward. However, by spending some time looking around, we can find some videos on the website funbase.cs.tut.fi
+Functionality documentation: API methods are documented (functionality, parameters, return value, etc.)
+Automated tests: the project has unit tests
-Community guidelines: When exiting the software, the user has the opportunity to complete a feedback form and submit it to the SW's developers.
This is nice (I would suggest keeping that window on screen for a bit longer). Also, in the SW itself, the "About" menu item contains an email address to contact the SW team.
However, I do not find how to contribute to the software. If it is indicated somewhere, it needs to be more obvious (eg in the README.md)
Also, the in-software help is poor. 

Software paper:
+Authors: The file paper.md states the authors and their affiliation
+A statement of need: A summary is included, but does not describe the target audience
+References: paper.md references two documents, both of which have a doi indicated.

**Comment by labarba** [comment]
Thank you for your review, @jflorence !!

@TermosPullo Back to you. Minor revisions, as described above. Let us know here when you think the revision is ready.

**Comment by TermosPullo** [comment]
@jflorence Hello again. We have now done the revisions. Your feed back was really useful for us! Moreover, reviewing a newer version than earlier is no problem to us. Please let us know if more is needed.

@labarba Could it be feasible to change the repository address of our submission? The current address is no longer functional and it appears I cannot change it myself.
The submission: http://joss.theoj.org/papers/73e33d6850d24f0d6aad0d5f38937f83
The functional address: https://github.com/kactus2/kactus2dev

**Comment by arfon** [comment]
> @labarba Could it be feasible to change the repository address of our submission?

@labarba - I've updated this.

**Comment by TermosPullo** [comment]
@arfon Thank you.

**Comment by labarba** [comment]
@jflorence The author reports completing the requested revisions. Are you satisfied with them to recommend acceptance? I still see five unchecked boxes in the review checklist.
Thanks!

**Comment by jflorence** [comment]
Sorry for late reply.
Everything is fine now, except for 
"Version: Does the release version given match the GitHub release (3.2.173)?"
What does it mean exactly?
The latest version I reviewed was 3.4.0

**Comment by arfon** [comment]
> The latest version I reviewed was 3.4.0

@jflorence - I've updated the release at the top of this issue to match.

**Comment by labarba** [comment]
@jflorence We would need your final statement that the submission is complete and ready to accept. Thanks for your review!

And check out the nice [feature on JOSS](http://blogs.nature.com/naturejobs/2017/04/04/joss-gives-computational-scientists-their-academic-due/) that appeared in Nature TechBlogs a few days ago.

**Comment by labarba** [comment]
@jflorence Allow me to ping again — The version was updated: if this is the last remaining issue, we await for you to check that box and to state here your recommendation for accepting the paper. Thanks!

**Comment by jflorence** [comment]
Yes, I recommend accepting the paper.

**Comment by arfon** [comment]
> Yes, I recommend accepting the paper.

@TermosPullo - At this point could you make an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive? I can then move forward with accepting the submission.

**Comment by TermosPullo** [comment]
@arfon Here: 10.5281/zenodo.570521

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.570521 as archive

**Comment by whedon** [comment]
OK. <a href="http://dx.doi.org/10.5281/zenodo.570521" target="_blank">10.5281/zenodo.570521</a> is the archive.

**Comment by arfon** [comment]
@jflorence many thanks for your review here and to @labarba for editing this submission ✨ 

@TermosPullo - your submission is now accepted into JOSS and your paper DOI is http://dx.doi.org/10.21105/joss.00151 ⚡️ 🚀 💥
